File: mvvm-test/src/main/java/com/martin/mvvm/repo/UserRepository.java
package com.martin.mvvm.repo;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.martin.mvvm.db.AppDatabase;
import com.martin.mvvm.db.bean.User;
import com.martin.mvvm.db.dao.UserDao;

import java.util.List;
import java.util.concurrent.Executor;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class UserRepository {
    private final UserDao userDao;
    private final LiveData<List<User>> userList;
    private final Executor executor = Executors.newSingleThreadExecutor();

    public UserRepository(Application application) {
        AppDatabase db = AppDatabase.getInstance(application);
        userDao = db.userDao();
        userList = userDao.getAllUsers();
    }

    public void insert(final User user) {
        executor.execute(() -> userDao.insertUser(user));
    }

    public LiveData<List<User>> getAllUsers() {
        return userList;
    }

    public void delete(final User user) {
        executor.execute(() -> userDao.deleteUser(user));
    }

    public void update(final User user) {
        executor.execute(() -> userDao.updateUser(user));
    }

    public void shutdown() {
        if (executor instanceof ExecutorService) {
            ExecutorService es = (ExecutorService) executor;
            es.shutdown();
        }
    }
}

File: room-livedata-recyclerview/src/main/java/com/example/rlr/UserRepository.kt
package com.example.rlr

import android.app.Application
import android.util.Log
import androidx.lifecycle.LiveData

class UserRepository(application: Application) {
    private val db: AppDatabase = AppDatabase.getInstance(application)
    private val userDao: UserDao = db.userDao()

    fun getUsers(): LiveData<List<User?>?>? {
        val data = userDao.getAllUsers()
        return data
    }

    fun insert(user: User) {
        Log.d("UserRepository", "insert user: $user")
        Thread {
            userDao.insert(user)
        }.start()
    }
}
